temporaltest/server_test.go

```diff
@@ -143,11 +152,17 @@ func TestNewWorkerWithOptions(t *testing.T) {
 	if err := wfr.Get(ctx, &result); err != nil {
 		t.Fatal(err)
 	}
-
 	if result != "Hello world" {
 		t.Fatalf("unexpected result: %q", result)
 	}
 
+	// Verify that the Identity worker option was set.
```

Comment:
Nice. Although, is this test going to be flaky because of partitions? I think DescribeTaskQueue only queries one. If you're willing to set some dynamic config here you could limit it to 1.

Reply:
Would this be controlled via `matching.numTaskqueue{Read,Write}Partitions` dynamic config?

Reply:
After more testing this doesn't seem to make a difference. Which I think actually makes sense, given that in this configuration the server will always have a single matching node.

Reply:
One node, yes, but it will still have multiple partitions loaded. Maybe it works reliably because of forwarding. Anyway, nothing to change if it passes.